source/concepts/protocols.rst

```diff
@@ -0,0 +1,284 @@
+   for the virtual channel is registered on-chain, the intermediary will try to
+   finalize the state of the sub-channel between the intermediary and the other
+   channel participant (who did not initiate settlement) through an off-chain
+   transaction. If is succeeds, then that sub-channel will be closed. If not,
+   then the ledger channel backing that sub-channel, along with all its
+   sub-channels will be registered on the blockchain.
+
+
+.. image:: ../_generated/concepts/register_generic_register.svg
+  :align: Center
+  :alt: Image not available
+
+After the challenge duration for register expires, if the channel has
+
+    1. ``No app``, then this state can be settled.
+    2. ``An app``, the channel transitions to force-execution sub-phase.
+
+3. Progress the channel state on the blockhain
+``````````````````````````````````````````````
+
+This is a special sub-phase of register, relevant only for channels that have
+an app.
+
+In this sub-phase, the pariticipants can update unanimously (using only their
+signatures) can update the state of the channel after the challenge duration
+for register has expired, but before the challenge duration for on-chain
+progression expires. Each time the update is proposed on-chain, app contract
+validates it and it it is valid, the state is updated.
+
+During each update, the challenge duration for on-chain progression is
+restarted. The pariticipants can make any number of on-chain progressions on
+the blockchain, before the the challenge duration expires. Once it expires, the
+channel progresses to the settle phase.
+
+.. image:: ../_generated/concepts/register_generic_progress.svg
+  :align: Center
+  :alt: Image not available
+
+.. note::
+
+    While the register phase is distinct in the protocol descriptions, it is up
+    to the imlementations to expose this phase to the user. For instance, in
+    `go-perun` implementation of these protocols, APIs for ``on-chain
+    progression`` and ``settle`` are exposed. Register is called implicitly by
+    both of these APIs.
+
+Settle phase
+------------
+
+In the settle phase, the state of the channel is concluded on the parent layer
+and the funds are withdrawn.
+
+1. For finalized sub-channels & virtual channels
+````````````````````````````````````````````````
+If a channel is a sub-channel or ledger channel and the state to be settled was
+finalized through off-chain transactions, then it could funds could be directly
+withdrawn by making an update to the parent layer. In case of
+
+1. Sub-channel: `parent layer` is the ledger channel between the particiants.
+2. Virtual channel: `parent layer` is the two ledger channels, one between each
+   of the pariticipants and the intermediary.
+
+The withdrawal protocol for sub-channel and virtual channel each of type of
+channel are described in detail in the ``Withdrawal protocols for sub-channels
+and virtual channels`` section.
```

Comment:
```suggestion
The withdrawal protocol for sub-channel and virtual channel
are described in detail in the ``Withdrawal protocols for sub-channels
and virtual channels`` section.
```
is it withdrawal protocol or protocols? --> should be consistent...

Reply:
Withdrawal protocols. I have updated it.